community/kernel/src/main/java/org/neo4j/kernel/api/ExecutingQuery.java

```diff
@@ -20,41 +20,64 @@
 package org.neo4j.kernel.api;
 
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+import org.neo4j.kernel.impl.locking.Locks;
 import org.neo4j.kernel.impl.query.QuerySource;
+import org.neo4j.storageengine.api.lock.ResourceType;
+import org.neo4j.time.CpuClock;
+import org.neo4j.time.SystemNanoClock;
 
-import static java.lang.String.format;
+import static java.util.concurrent.atomic.AtomicLongFieldUpdater.newUpdater;
 
 /**
  * Represents a currently running query.
  */
 public class ExecutingQuery
 {
+    private static final AtomicLongFieldUpdater<ExecutingQuery> WAIT_TIME =
+            newUpdater( ExecutingQuery.class, "waitTimeNanos" );
     private final long queryId;
+    private final Locks.Tracer lockTracer = ExecutingQuery.this::waitForLock;
```

Comment:
The place where a system-configured `Locks.Tracer` should be injected is in `KernelStatement` in the `lockTracer()` method, instead of the hard-coded `Locks.Tracer.NONE` there.